include/seqan3/std/span.hpp

```diff
@@ -0,0 +1,551 @@
+    constexpr const_reverse_iterator crend() const noexcept;
+
+private:
+    pointer data_;     // exposition only
+    index_type size_;  // exposition only
+};
+
+template<class T, size_t N>
+    span(T (&)[N]) -> span<T, N>;
+
+template<class T, size_t N>
+    span(array<T, N>&) -> span<T, N>;
+
+template<class T, size_t N>
+    span(const array<T, N>&) -> span<const T, N>;
+
+template<class Container>
+    span(Container&) -> span<typename Container::value_type>;
+
+template<class Container>
+    span(const Container&) -> span<const typename Container::value_type>;
+
+} // namespace std
+
+*/
+
+// #include <__config>
+#include <cstddef>      // for ptrdiff_t
+#include <iterator>     // for iterators
+#include <array>        // for array
+#include <type_traits>  // for remove_cv, etc
+#include <cstddef>      // for byte
+
+#pragma once
+
+#if __cplusplus <= 201703L
+
+namespace std
+{
+inline constexpr ptrdiff_t dynamic_extent = -1;
+template <typename span_tp, ptrdiff_t span_extent = dynamic_extent> class span;
+
+
+template <class span_tp>
+struct is_span_impl : public false_type {};
+
+template <class span_tp, ptrdiff_t span_extent>
+struct is_span_impl<span<span_tp, span_extent>> : public true_type {};
+
+template <class span_tp>
+struct is_span : public is_span_impl<remove_cv_t<span_tp>> {};
+
+template <class span_tp>
+struct is_std_array_impl : public false_type {};
+
+template <class span_tp, size_t span_sz>
+struct is_std_array_impl<array<span_tp, span_sz>> : public true_type {};
+
+template <class span_tp>
+struct is_std_array : public is_std_array_impl<remove_cv_t<span_tp>> {};
+
+template <class span_tp, class ElementType, class = void>
+struct is_span_compatible_container : public false_type {};
+
+template <class span_tp, class ElementType>
+struct is_span_compatible_container<span_tp, ElementType,
+        void_t<
+        // is not a specialization of span
+            typename enable_if<!is_span<span_tp>::value, nullptr_t>::type,
+        // is not a specialization of array
+            typename enable_if<!is_std_array<span_tp>::value, nullptr_t>::type,
+        // is_array_v<Container> is false,
+            typename enable_if<!is_array_v<span_tp>, nullptr_t>::type,
+        // data(cont) and size(cont) are well formed
+            decltype(data(declval<span_tp>())),
+            decltype(size(declval<span_tp>())),
+        // remove_pointer_t<decltype(data(cont))>(*)[] is convertible to ElementType(*)[]
+            typename enable_if<
+                is_convertible_v<remove_pointer_t<decltype(data(declval<span_tp &>()))>(*)[],
+                                 ElementType(*)[]>,
+                nullptr_t>::type
+        >>
+    : public true_type {};
+
+
+template <typename span_tp, ptrdiff_t span_extent>
+class span {
+public:
+//  constants and types
+    using element_type           = span_tp;
+    using value_type             = remove_cv_t<span_tp>;
+    using index_type             = ptrdiff_t;
```

Comment:
this has now been changed to `size_t` so it's unsigned. We should change it in our version, too.